addons/account/models/account_move.py

```diff
@@ -1818,12 +1818,14 @@ def _create_tax_cash_basis_base_line(self, amount_dict, amount_currency_dict, ne
                 base_line,
                 debit=rounded_amt > 0 and rounded_amt or 0.0,
                 credit=rounded_amt < 0 and abs(rounded_amt) or 0.0,
-                amount_currency=amount_currency))
+                amount_currency=amount_currency,
+                journal_id=new_move.journal_id.id))
             aml_obj.create(dict(
                 base_line,
                 credit=rounded_amt > 0 and rounded_amt or 0.0,
                 debit=rounded_amt < 0 and abs(rounded_amt) or 0.0,
                 amount_currency=-amount_currency,
+                journal_id=new_move.journal_id.id,
```

Comment:
I'd rather add journal_id once in _get_tax_cash_basis_base_common_vals() instead of two times here; for the sake of maintaining the genericity of this function (which is designed as a hook anyway, so it's better to do it there).